### PR TITLE
Bind column charts to api

### PR DESCRIPTION
**Pull request by pixelfact**

### CODE REVIEW

File: app/programs/[programId]/_features/financial-column-chart/financial-column-chart.hooks.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import { bootstrap } from "@/core/bootstrap";
+import { GetProgramTransactionsStatsResponse } from "@/core/domain/program/program-contract.types";
+
+import { Typo } from "@/design-system/atoms/typo";
+
+export function useFinancialColumnChart(stats?: GetProgramTransactionsStatsResponse["stats"]) {
+  const { format: formatDate } = bootstrap.getDateKernelPort();
+  const { format, getCurrency } = bootstrap.getMoneyKernelPort();
+
+  const categories = stats?.map(stat => formatDate(new Date(stat.date), "MMMM yyyy")) ?? [];
```
Comment (NeoxAzrot) [inline app/programs/[programId]/_features/financial-column-chart/financial-column-chart.hooks.tsx:12]:
```suggestion
  const categories = stats?.map(stat => dateKernelPort.format(new Date(stat.date), "MMMM yyyy")) ?? [];
```

File: app/programs/[programId]/_features/financial-column-chart/financial-column-chart.hooks.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  const receivedSeries = calculateSeries("totalAvailable");
+  const grantedSeries = calculateSeries("totalGranted");
+  const rewardedSeries = calculateSeries("totalRewarded");
+
+  const renderAmount = useCallback(
+    (amountSum: number) => {
+      return (
+        <div className="flex gap-1">
+          <Typo size={"xs"} color={"text-1"}>
+            {format({ amount: amountSum, currency: getCurrency("USD") }).amount}
```
Comment (NeoxAzrot) [inline app/programs/[programId]/_features/financial-column-chart/financial-column-chart.hooks.tsx:35]:
```suggestion
            {moneyKernelPort.format({ amount: amountSum, currency: moneyKernelPort.getCurrency("USD") }).amount}
```

File: app/programs/[programId]/_features/financial-column-chart/financial-column-chart.hooks.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+          </Typo>
+          <Typo size={"xs"} color={"text-2"}>
+            {format({ amount: amountSum, currency: getCurrency("USD") }).code}
+          </Typo>
+        </div>
+      );
+    },
+    [format, getCurrency]
```
Comment (NeoxAzrot) [inline app/programs/[programId]/_features/financial-column-chart/financial-column-chart.hooks.tsx:43]:
```suggestion
    [moneyKernelPort]
```

File: design-system/atoms/chart-legend/chart-legend.loading.tsx [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export function ChartLegendLoading() {
+  return <div />;
```
[the remark this answers is not in the document]
Comment (pixelfact) [reply design-system/atoms/chart-legend/chart-legend.loading.tsx:2]: i think we don't need to have a loading component for this atom. Deleted !

File: app/programs/[programId]/_features/financial-column-chart/financial-column-chart.tsx [hunk 5]
```diff
@@ -1,157 +1,93 @@
...
+          <div className="flex justify-between gap-4">
+            <ChartLegend colors="container-3">
+              <Translate token={"programs:financialColumnChart.legends.rewarded"} />
+            </ChartLegend>
+            {renderRewardedAmount}
+          </div>
+        </Paper>
+        {/*TODO @Mehdi handle date range change*/}
+        <div>Date Range popover trigger</div>
+      </div>
```
[the remark this answers is not in the document]
Comment (pixelfact) [reply app/programs/[programId]/_features/financial-column-chart/financial-column-chart.tsx:90]: by extracting the logic into a custom hook, the component remain simple and avoid props drilling

File: shared/components/empty-state/empty-state.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  actionLabelTranslate,
+  onAction,
+}: EmptyStateProps) {
+  const Component = as;
+
+  return (
+    <Component className="flex flex-col items-center justify-center gap-6 p-6 text-center">
+      <div className="w-20">
+        <Image src={Categories} width={80} height={80} alt={"emptyStatePictureFallback"} />
+      </div>
```
[the remark this answers is not in the document]
Comment (pixelfact) [reply shared/components/empty-state/empty-state.tsx:33]: this component is (dirty) temp, we need to add a clean component in DS

File: design-system/atoms/chart-legend/adapters/default/default.adapter.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+}: ChartLegendPort<C>) {
+  const Component = as || "div";
+  const slots = ChartLegendDefaultVariants({ size, colors });
+  return (
+    <Component {...htmlProps} className={cn(slots.base(), classNames?.base)}>
+      <div className={cn(slots.square(), classNames?.square)}></div>
+
+      <Typo
+        as={"div"}
+        size={size === "m" ? "xs" : "xxs"}
```
[the remark this answers is not in the document]
Comment (pixelfact) [reply design-system/atoms/chart-legend/adapters/default/default.adapter.tsx:26]: its the chart legend size

**Review by NeoxAzrot — APPROVED** [review]

**Pull request merged**